HUDOC.js

```diff
@@ -0,0 +1,502 @@
+		&& !(text(doc, "title").toLowerCase().includes("translation]") // toLowerCase() is added because "translation" is sometimes capitalized
+			|| docType.includes("Communiqué de presse"))) {
+			return "case";
+		}
+	}
+	else if (url.includes("hudoc.echr.coe.int/eng#")) {
+		//English website (so won't work for Spanish or German)
+		if ((docType.includes("Judgment")
+			|| docType.includes("Decision")
+			|| docType.includes("Advisory Opinion")
+			|| docType.includes("Res-")
+			|| docType.includes("Communicated"))
+		// Exclude translations and press releases.
+		&& !(text(doc, "title").toLowerCase().includes("translation]") // toLowerCase() is added because "translation" is sometimes capitalized
+			|| docType.includes("Press Release"))) {
+			return "case";
+		}
+	}
+
+
+	return false;
+}
+
+function doWeb(doc, url) {
+	if (detectWeb(doc, url) == "case") {
+		scrapeDecision(doc, url);
+	}
+}
+
+function scrapeDecision(doc, url) { // Works for both Court judgments and decisions
+	var item = new Zotero.Item("case");
+
+	// Title
+	var capTitle = ZU.capitalizeTitle(text(doc, "title").toLowerCase(), true);
+
+	var splitTitle = capTitle.split(/\(/);
+
+	// If title already contains the type, remove it.
+	// Exclude non-split titles and advisory opinions
+	if (!(splitTitle.length == 1) &&
+			!scrapeMetaData(doc, "typedescription").includes("Advisory Opinion")) {
+
+		let titleVersusMatch = splitTitle.findIndex(function(elem) {
+			if (elem.includes("V.") || elem.includes("C.")) {
+						return true;
+			}
+			return false;
+		});
+
+		let titleNumMatch = splitTitle.findIndex(function(elem) {
+			if (/^(no.|II|IV|V\)|VI|IX|X\))/i.test(elem)) {
+						return true;
+			}
+			return false;
+		});
+
+		let titleIndex = Math.max(titleVersusMatch, titleNumMatch);
+
+
+		capTitle = splitTitle.slice(0, titleIndex+1).join("(");
+	}
+
+	// Zotero capitalizes the "v."/"c.", so we need to correct that for English and French cases
+	if (capTitle.includes(" V. ")) {
+		capTitle = capTitle.replace(" V. ", " v. ");
+	}
+	else if (capTitle.includes(" C. ")) {
+		capTitle = capTitle.replace(" C. ", " c. ");
+	}
+	item.caseName = capTitle + getTypeBit(doc, url);
+
+	// Court
+	var court = scrapeMetaData(doc, "originatingbody");
+
+	if (url.includes("hudoc.echr.coe.int/fre#")) {
+		item.court = "ECtHR";
+	}
+	else {
+		if (text(doc, "title").includes("Advisory Opinion")) {
+			item.court = "ECtHR [GC]";
+		}
+		else if (court.includes("Grand Chamber")) {
+			item.court = "ECtHR [GC]";
+		}
+		else if (court.includes("Commission")) {
+			item.court = "Commission";
+		}
+		else if (court == "Committee of Ministers") {
+			item.court = "Committee of Ministers";
+		}
+		else {
+			item.court = "ECtHR";
+		}
+	}
+
+	// Date of decision
+	// convert to simple ISO: yyyy-mm-dd dd.mm.yyyy.
+	item.dateDecided = scrapeMetaData(doc, "judgementdate").split("/").reverse().join('-');
+
+	item.url = "http://hudoc.echr.coe.int/eng?i=" + getItemID(url);
```

Comment:
Is it right to always link to the English site? Also should use `https://`.

Reply:
Fixed